mathics/builtin/strings.py

```diff
@@ -638,6 +638,69 @@ class LetterCharacter(Builtin):
      = True
     """
 
+# FIXME: Generalize string.lower() and ord()
+def letter_number(chars: List[str], start_ord) -> List['Integer']:
+    # Note caller has verified that everything isalpha() and
+    # each char has length 1.
+    return [Integer(ord(char.lower()) - start_ord) for char in chars]
+
+class LetterNumber(Builtin):
+    """
+    <dl>
+      <dt>'LetterNumber'[$c$]
+      <dd>returns the position of the character $c$ in the English alphabet.
+
+      <dt>'LetterNumber["string"]'
```

Comment:
I would implement `Alphabet[]` at least for English, (Latin?) and the second parameter, in a way that anything that is not the default raises a message. At the end, the basic implementation of Alphabet is  just a dictionary with the form "alphabetname": "abcd..."
isn't it?

Reply:
You have a clearer idea of what to do at this point than I do. So if you'd just do it, I'd appreciate. 

Otherwise I am happy to have this hang out as a draft for a while. 

For larger context, I have been going over https://www.wolfram.com/language/elementary-introduction/2nd-ed/11-strings-and-text.html and more generally https://www.wolfram.com/language/elementary-introduction/2nd-ed/ which has the attribution Copyright 2021 (and nothing more). The 1st Edition, which I have in book form, has a "non-commercial share-alike" copyright and that is similar. 

And the bigger context even here is that these give examples that can be used in worksheets. (See the docker image or the sqlite file in mathics-omnibus and please also see https://github.com/Mathics3/mathics-django/pull/32).  But in addition this is a much more gentle way to guide us in filling out the code in a way where users can see immediate results. 

The problem I have with FeynCalc, Rubi, KnotTheory and similar packages like this is that they are hundreds if not thousands of lines long and use sometimes sophisticated constructs in intertwined ways.  I think if we have a more solid base to start out with, things will go easier there. 

I tried pulling a 2006 version of KnotTheory and tried building it. In terms of the number of things we need to fix, there are far fewer. I counted maybe 3 or 4 things. Still, without those I am not able to get this to run. And at least for me this is kind of disappointing.

Reply:
>  At the end, the basic implementation of Alphabet is just a dictionary with the form "alphabetname": "abcd..."
> isn't it?
I think there exists python bindings for the ICU project to create alphabets:
- http://site.icu-project.org/

Unfortunately the alphabets doesn't seem to be exactly the same as in MMA.

Reply:
I looked at the ICU project and it looks both awesome and bureaucratic. 

The desire is extremely simple and basic: for each language give me an ordered list of the alphabet with a way to convert from one case to the other. And don't even need bidirectional conversion so you can choose which case to start out with.

However, although I can see how to do case conversion using unicode properties (and you'd think then that a library would use that provide such a function equivalent to "lower" or "upper"), there isn't anything that says give me the first letter in the alphabet and iterate to to the last letter, as far as I can tell. 

But I could be wrong here. The documentation while extensive isn't all that useful. Sigh.

Reply:
I think that an ICU-based `Alphabet` is something to implement in an external module. Here, I would limit to define a basic set of alphabets (let's say,  "English", "Spanish", "German", and  "Greek"  alphabets, that is what I could handle). A Pymathics module then can overload this basic implementation.

Reply:
With ICU more things like `IntegerName` and `Transliterate` can be created:
- https://github.com/axkr/symja_android_library/blob/master/symja_android_library/doc/functions/IntegerName.md
- https://github.com/axkr/symja_android_library/blob/master/symja_android_library/doc/functions/Transliterate.md

Reply:
Yes, I was thinking along those lines too. There seems a lot of functionality somewhere in there, which extends beyond just Alphabet.

At some point I will post a query on StackOverflow. But if you look at past queries on this topic they generally are met with derision. I just upvoted https://stackoverflow.com/questions/32375797/what-unicode-ranges-are-considered-letters and it appears I am the _only_ one to have done so just now after 5 1/2 years with no great answers.

Reply:
NO problemif you want to port it.